integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java

```diff
@@ -164,14 +165,26 @@ public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offers) {
           targetCpu = workerCpu;
           targetMem = workerMem;
           mWorkers.add(offer.getHostname());
+          mIsMasterTask = false;
         } else {
           // The resource offer cannot be used to start either master or a worker.
+          driver.declineOffer(offer.getId());
           continue;
         }
+        
+        mTaskName = mIsMasterTask ? sConf.get(Constants.INTEGRATION_MESOS_TACHYON_MASTER_NAME) : 
+           sConf.get(Constants.INTEGRATION_MESOS_TACHYON_WORKER_NAME);
+        Protos.TaskID taskId =
+                Protos.TaskID.newBuilder().setValue(mTaskName + mLaunchedTasks).build();
```

Comment:
I think, we must use mTaskName + mLaunchedTasks because in statusUpdate we are controlling based on taskID which contains tachyon_master_name